ballerina/vector.bal

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) 2023 WSO2 LLC. (http://www.wso2.com) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+
+# Calculates the l1 and l2 norm of a float vector.
+#
+# ```ballerina
+# float[] vector1 = [1.0, 2.0, 3.0];
+# vectorNorm(vector1, "l2") ⇒ 3.7416573867739413
+# vectorNorm(vector1, "l1") ⇒ 6.0
+# ```
+#
+# + vector1 - float vector to operate on
+# + norm - norm type to calculate out of ["l1", "l2"]
+# + return - the l1 or l2 norm of the given float vector
+function vectorNorm(float[] vector1, string norm = "l2") returns float {
+
+    if norm == "l2" {
+        float magnitudeSquared = 0.0;
+        foreach int i in 0 ..< vector1.length(){
+            magnitudeSquared += (<float>vector1[i]).pow(2);
+        }
+        return (<float>magnitudeSquared).sqrt();
+    }
+    else if norm == "l1" {
```

Comment:
Incorrect formatting. (If we use an `enum` for the `norm`, we can omit the condition here as the `else` block should be `l1` after the first `if` block.

```suggestion
    } else if norm == "l1" {
```

Reply:
Addressed in https://github.com/ballerina-platform/module-ballerina-math.vector/pull/1/commits/1fe0f828ef1ea29a4fc27924af5f541af69729ff.